homeassistant/components/gardena_bluetooth/__init__.py

```diff
@@ -0,0 +1,90 @@
+"""The Gardena Bluetooth integration."""
+from __future__ import annotations
+
+import asyncio
+import logging
+
+from bleak.backends.device import BLEDevice
+from gardena_bluetooth.client import CachedConnection, Client
+from gardena_bluetooth.const import DeviceConfiguration, DeviceInformation
+from gardena_bluetooth.exceptions import CommunicationFailure
+
+from homeassistant.components import bluetooth
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_ADDRESS, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.entity import DeviceInfo
+import homeassistant.util.dt as dt_util
+
+from .const import DOMAIN
+from .coordinator import Coordinator
+
+PLATFORMS: list[Platform] = [Platform.SWITCH]
+LOGGER = logging.getLogger(__name__)
+TIMEOUT = 20.0
+DISCONNECT_DELAY = 5
+
+
+class DeviceUnavailable(CommunicationFailure):
+    """Raised if device can't be found."""
+
+
+def get_connection(hass: HomeAssistant, address: str) -> CachedConnection:
+    """Set up a cached client that keeps connection after last use."""
+
+    def _device_lookup() -> BLEDevice:
+        device = bluetooth.async_ble_device_from_address(
+            hass, address, connectable=True
+        )
+        if not device:
+            raise DeviceUnavailable("Unable to find device")
+        return device
```

Comment:
Does this need to raise? Can it return `None` instead?

Reply:
Didn't really need to subclass, made it a home assistant exception instead.